weave-js/src/components/PagePanelComponents/Home/Browse3/datasets/DatasetEditorContext.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {createContext, useCallback, useContext, useState} from 'react';
+
+interface DatasetRow {
+  id: string | number;
+  [key: string]: any;
+}
+
+interface EditedCell {
+  [fieldName: string]: unknown;
+}
+
+interface DatasetEditContextType {
```

Comment:
A few questions here:

1. I am assuming that added rows are always either added to the end?
2. I am assuming that deleted rows are always against the original dataset (eg. you never "delete an added row")

If these assumptions hold, then edit indexes are stable.

Next:
maybe this will be clear below, but having two edit maps is odd to me. i am not sure I understand why this is needed and seems prone to potential bugs.

Reply:
I consolidated the edit maps, that was a very confusing choice (by me).